### DISCUSSION TITLE
User merge and block account in Diksha.

### DISCUSSION THREAD

**Discussion by rupeshpyeole** [Sunbird-Lern/Community#170] [Issues] [answered]
Hi Team, 
We want to know how user merge takes place in diksha - Means, If user merge his previous account with new account , How can we find his new account. As we are getting his previous account as **isdeleted** status "True".
Due to this issue some users didn't received certificate as their account has been blocked in database, Hence, we want know how can we find his new account and how can we track his telemetry events with previous Id and new Id if it is generated. 

We kindly request you to please provide inputs on this issue.

Thanks,
Rupesh

**Comment by Hari-stackroute** [comment]
Hi @rupeshpyeole
User Merge scenario is merging state user account(non-custodian) with his custodian account, So for this user usually login as diksha user(state-user) and merges the non-state user account. Usually courses and certificates everything merges from non-state user account to diksha account. There is no point of new account here, without login to diksha account merge scenario is not at all possible.
Please check for this telemetry log for finding the details:
`UserMergeActor:triggerUserMergeTelemetry: generating telemetry event for merge`

cc: @AmiableAnil

**Reply by rupeshpyeole** [reply]
Hi Hari,
We really appreciate your kind reply. However, we would like to know following details.
1.Is the merge account details gets stored separately in database?
 If yes, then in which database and table.
2. once custodian account has been merged with state account then custodian account has gone in blocked status. There is any flag in DB to identify this is merge account. 
3.There is any linking/relation b/w custodian account with state account (merge account) at DB level.

**we are not able to find telemetry events for the merge users in druid data sources, please suggest in which DataSource can we find it if they are there. 

Thanking You,
Rupesh

**Reply by Jayaprakash8887** [reply, marked as answer]
@rupeshpyeole No field is capturing the details of merged account. 
1. service logs 
2. Telemetry audit event. 
3. Check the kafka event lag of https://lern.sunbird.org/use/developer-guide/lms-service/lms-flink-jobs/merge-user-courses

**Reply by Jayaprakash8887** [reply]
@rupeshpyeole Please let us know if we can mark the thread as answered if there are no further queries.

**Reply by rupeshpyeole** [reply]
Hi @Jayaprakash8887 Presently we don't have any further queries if we have in future will resume discussion.

Thank you so much for the answering to our queries.

Thanks,
Rupesh
